cmd/ls/ls_test.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var tempDir string
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	teardown()
+	os.Exit(code)
+}
+
+func setup() {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		fatal(err)
+	}
+
+	tempDir = dir
+	createTempFiles(tempDir)
```

Comment:
A cool hack in Go is do something like this:

```
func setup() func() {

      // Do all your shit
      return func() {
              // Tear down your shit using a closure
      }
}
```

Then on test code you can do:

```
func TestLalala(t *testing.T) {
    teardown := setup(t)
    defer teardown()
}
```

or 

```
func TestLalala(t *testing.T) {
    defer setup(t)()
}
```